File: 0661-image-smoother/0661-image-smoother.cpp
class Solution {
public:
    vector<vector<int>> imageSmoother(vector<vector<int>>& img) {
        int dir[9][2] = {{-1,-1},{-1,0},{-1,1},{0,-1},{0,0},{0,1},{1,-1},{1,0},{1,1}};
        vector<vector<int>> ans;
        for(int r = 0; r < img.size(); r++){
            vector<int> tmp;
            int cur_r, cur_c;    
            for(int c = 0; c < img[r].size(); c++){
                int cnt = 0, sum = 0;
                for(int i = 0; i < 9; i++){
                    cur_r = r + dir[i][0];
                    cur_c = c + dir[i][1];
                    if(cur_r < 0 || cur_c < 0 || cur_r >= img.size() || cur_c >= img[0].size()){
                        continue;
                    }
                    cnt++;
                    sum += img[cur_r][cur_c];
                }
                tmp.push_back(round(sum / cnt));
            }
            ans.push_back(tmp);
        }
        return ans;
    }
};

File: 0661-image-smoother/0661-image-smoother.cs
public class Solution {
    public int[][] ImageSmoother(int[][] img) {
        //  Editorial
        int m = img.Length;
        int n = img[0].Length;

        int[] temp = new int[n];
        int prevVal = 0;

        for(int i = 0; i < m; i++){
            for(int j = 0; j < n; j++){
                int sum = 0;
                int count = 0;

                //  아래 이웃
                if(i + 1 < m){
                    if(j - 1 >= 0){
                        sum += img[i + 1][j - 1];
                        count += 1;
                    }
                    sum += img[i + 1][j];
                    count += 1;
                    if(j + 1 < n){
                        sum += img[i + 1][j + 1];
                        count += 1;
                    }
                }

                //  옆 이웃(현재 위치에서 오른쪽)
                if(j + 1 < n){
                    sum += img[i][j + 1];
                    count +=1;
                }

                //  현재 위치
                sum += img[i][j];
                count += 1;

                //  옆 이웃(현재 위치에서 왼쪽)
                if(j - 1 >= 0){
                    sum += temp[j - 1];
                    count += 1;
                }

                //  위 이웃
                if(i - 1 >= 0){
                    //  
                    if(j - 1 >= 0){
                        sum += prevVal;
                        count += 1;
                    }

                    sum += temp[j];
                    count += 1;

                    if(j + 1 < n){
                        sum += temp[j + 1];
                        count += 1;
                    }
                }

                if(i - 1 >= 0){
                    prevVal = temp[j];
                }

                temp[j] = img[i][j];

                img[i][j] = sum / count;
            }
        }

        return img;
    }
}